kratos/spatial_containers/bins_dynamic_objects.h

```diff
@@ -107,21 +108,25 @@ class BinsObjectDynamic {
   BinsObjectDynamic (IteratorType const& ObjectsBegin, IteratorType const& ObjectsEnd)
       : mObjectsBegin(ObjectsBegin), mObjectsEnd(ObjectsEnd) {
 
+    std::size_t GridSize[3] = {0, 0, 0};
+
     mObjectsSize = SearchUtils::PointerDistance(mObjectsBegin,mObjectsEnd);
-    CalculateBoundingBox();           // Calculate mMinPoint, mMaxPoint
-    CalculateCellSize(mObjectsSize);  // Calculate number of Cells
-    AllocateContainer();              // Allocate cell list
-    GenerateBins();                   // Fill Cells with objects
+    CalculateBoundingBox();                                                 // Calculate mMinPoint, mMaxPoint
+    CalculateCellSize(mObjectsSize, GridSize);                              // Calculate number of Cells
```

Comment:
Please leave this one without additional parameter. It doesn't needed internally and you don't need to invent it.

Reply:
I think GridSize is the actual number of cells, not its size. If that is the case I cannot use it like you say. Right now CellSize indicates the size of the cell, no the number of cells.

That's the reason the two constructors are changed